src/parser/if-then-else.js

```diff
@@ -4,47 +4,53 @@ import type Context from "./context";
 import statement from "./statement";
 import expression from "./expression";
 
-const ifThenElse = (ctx: Context) => {
-  const node = {
-    ...ctx.startNode(ctx.token),
-    then: [],
-    else: []
-  };
-  ctx.eat(["if"]);
-  // First operand is the expression
+// push statement based on then/else branch
+const pushStatement = (node, isThenBranch, stmt) => {
+  if (!stmt) return;
+  if (isThenBranch) return node.then.push(stmt);
+  return node.else.push(stmt);
+}
+
+const doIfExpression = (node, ctx) => {
   ctx.expect(["("]);
   node.expr = expression(ctx, "i32");
   ctx.expect([")"]);
+}
 
-  // maybe a curly brace or not
+// push statements while taking into consideration having curly braces or not
+const doStatement = (node, isThenBranch, ctx) => {
```

Comment:
I don't love how we pass through `isThenBranch` and `node` into this function. Mainly because it's not doing anything with the flags, it simply passes them through to `doIfExpression`. These type of patterns are a code smell that a function is doing a bit too much.

I like that you split this out into its own function and I think `doStatement` should just eat as many statements as it can and return them in an array.

```js
const doStatement = (ctx: Context) => {
   const statements = [];   
   if (ctx.eat(["{"])) {
      while(...) {
         statements.push(statement(ctx));
      }
      ...
   } else {
      statements = statement(ctx);
     ...
   }

   // filter out null statments, these should really be fixed in general :/
   return statements.filter(stmt => stmt != null);
}
```